packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1146,6 +1208,12 @@ export class GarbageCollector implements IGarbageCollector {
                 );
             } else {
                 nodeStateTracker.updateTracking(currentReferenceTimestampMs);
+                if (this.tombstoneMode && nodeStateTracker.state === UnreferencedState.SweepReady) {
+                    const nodeType = this.runtime.getNodeType(nodeId);
+                    if (nodeType === GCNodeType.DataStore || nodeType === GCNodeType.Blob) {
```

Comment:
This logic is duplicated in my code, we should do something to consolidate the logic :)